### PR TITLE
new(ci): reusable workflows

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind cleanup

**Any specific area of the project related to this PR?**

/area build

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by FedeDP** [comment]
I was able to test that download artifacts (before publishing) actually worked: https://github.com/falcosecurity/plugins/actions/runs/4945583976/jobs/8842857629
I will rebase this one to properly test and we should be good to go.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    branches: [ master ]
+
+# Checks if any concurrent jobs under the same pull request or branch are being executed
+# NOTE: this will cancel every workflow that is being ran against a PR as group is just the github ref (without the workflow name)
+concurrency:
+  group: ${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  build-plugins-ci:
```
Comment (FedeDP) [inline .github/workflows/ci.yaml:14]: CI on PR only tests the build.
Note: it also uses `reusable_build_packages` workflow from HEAD.

File: .github/workflows/master.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: Update Plugins-dev
+
+on:
+  push:
+    branches: [ master ]
+  workflow_dispatch:
+
+# Checks if any concurrent jobs is already being executed for master and cancel it.
+concurrency:
+  group: ci-master
```
Comment (FedeDP) [inline .github/workflows/master.yaml:10]: Avoid sync problem with multiple jobs on master.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -34,90 +25,22 @@ jobs:
...
-        with:
-          role-to-assume: "arn:aws:iam::292999226676:role/github_actions-plugins-s3"
-          aws-region: ${{ env.AWS_S3_REGION }}
-
-      - name: Upload files to S3 ⬆️
-        run: |
-          for package in output/*.tar.gz; do
-            aws s3 cp --no-progress $package s3://${{ env.AWS_S3_BUCKET }}/${{ env.AWS_S3_PREFIX }}/stable/
-          done
+      - name: Fail on non semver tag
```
Comment (FedeDP) [inline .github/workflows/release.yml:29]: No need to go further.

**Review by jasondellaluce — APPROVED** [review]

**Pull request merged**